### DISCUSSION TITLE
Falta arquivos do Modulo -  Acesso à dados com .NET, C#, Dapper e SQL Server (57%)

### DISCUSSION THREAD

**Discussion by etaptecnologia** [balta-io/forum#132] [Tira Dúvidas] [answered]
@andrebaltieri 
Boa tarde.

No repositório não se encontra o banco de dados, nem as querys para agilizar o processo de criação do banco.

Poderia atualizar o repositório por favor?

https://github.com/balta-io/2806

Obrigado!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @etaptecnologia, como vai?
Este curso é uma continuação da carreira .NET, e ele segue depois do curso de SQL Server.... No curso de SQL Server criamos estes bancos, recomendo fazer ele também.

O repositório do curso é este aqui:
https://github.com/balta-io/2805

**Reply by etaptecnologia** [reply]
Não encontrei o DB da aula mensionada no link que foi enviado, inclusive assisti todas as essas aulas.

Outra coisa interessante, que nessa aula (Criando as tabelas) modulo (Acesso à dados com .NET, C#, Dapper e SQL Server), você não apresenta a tabela User.

Como eu nao encontrei o arquivo mensionado, eu fui criar 1 por 1 de acordo com a aula, pois faltou falar sobre o User.

**Reply by andrebaltieri** [reply]
@etaptecnologia anotei aqui para dar uma revisada... muito obrigado pelo feedback... se puder exportar seu banco e mandar aqui, pode ajudar outros alunos <3

**Reply by etaptecnologia** [reply]
CREATE TABLE [Role](
    [Id] INT NOT NULL IDENTITY(1,1),
    [Name] VARCHAR(80) NOT NULL,
    [Slug] VARCHAR(80) NOT NULL,

    CONSTRAINT [PK_Role] PRIMARY KEY([Id]),
    CONSTRAINT [UQ_Role_Slug] UNIQUE([Slug]),
)

CREATE NONCLUSTERED INDEX [IX_Role_Slug] ON [Role]([Slug])



CREATE TABLE [User](
    [Id] INT NOT NULL IDENTITY(1,1),
    [Name] VARCHAR(80) NOT NULL,
    [Email] VARCHAR(80) NOT NULL,

    CONSTRAINT [PK_User] PRIMARY KEY([Id]),
    CONSTRAINT [Email] UNIQUE([Email]),
)
CREATE NONCLUSTERED INDEX [IX_User_Email] ON [User]([Email])


CREATE TABLE [UserRole](
    [UserId] INT NOT NULL,
    [RoleId] INT NOT NULL,

    CONSTRAINT [PK_UserRole] PRIMARY KEY([UserId], [RoleId])
)


CREATE TABLE [Category](
    [Id] INT NOT NULL IDENTITY(1, 1),
    [Name] VARCHAR(80) NOT NULL,
    [Slug] VARCHAR(80) NOT NULL

    CONSTRAINT [PK_Category] PRIMARY KEY([Id]),
    CONSTRAINT [UQ_Category_Slug] UNIQUE([Slug])
)

CREATE NONCLUSTERED INDEX [IX_Category_Slug] ON [Categoru]([Slug])


CREATE TABLE [Post](
    [Id] INT NOT NULL IDENTITY(1,1),
    [CategoryId] INT NOT NULL,
    [AuthorId] INT NOT NULL,
    [Title] VARCHAR(160) NOT NULL,
    [Summary] VARCHAR(255) NOT NULL,
    [Body] TEXT NOT NULL,
    [Slug] VARCHAR(80) NOT NULL,
    [CreateDate] SMALLDATETIME NOT NULL DEFAULT(GETDATE()),
    [LastUpdateDate] SMALLDATETIME NOT NULL DEFAULT(GETDATE()),

    CONSTRAINT [PK_Post] PRIMARY KEY([Id]),
    CONSTRAINT [FK_Post_Category] FOREIGN KEY([CategoryId]) REFERENCES [Category]([Id]),
    CONSTRAINT [FK_Post_Author] FOREIGN KEY([AuthorId]) REFERENCES [User]([Id]),
    CONSTRAINT [UQ_Post_Skug] UNIQUE([Slug])
)

CREATE NONCLUSTERED INDEX [IX_Post_Slug] ON [Post]([Slug])



CREATE TABLE [Tag](
    [Id] INT NOT NULL IDENTITY(1,1),
    [Name] VARCHAR(80) NOT NULL,
    [Slug] VARCHAR(80) NOT NULL,

    CONSTRAINT [PK_Tag] PRIMARY KEY([Id]),
    CONSTRAINT [UQ_Tag_Slug] UNIQUE([Slug])
 )

 CREATE NONCLUSTERED INDEX [IX_Tag_slug] ON [Tag]([Slug])


 CREATE TABLE [PostTag](
    [PostId] INT NOT NULL,
    [TagId] INT NOT NULL,

    CONSTRAINT PK_PostTag PRIMARY KEY ([PostId], [TagId])
)
